File: 0057-insert-interval/0057-insert-interval.js
/**
 * @param {number[][]} intervals
 * @param {number[]} newInterval
 * @return {number[][]}
 */
var insert = function(intervals, newInterval) {
    const n = intervals.length;

    const ans = new Array();
    let i = 0;

    while(i < n && intervals[i][1] < newInterval[0]){
        ans.push(intervals[i++]);
    }
        

    while(i < n && intervals[i][0] <= newInterval[1]){

        newInterval[0] = Math.min(newInterval[0], intervals[i][0]);
        newInterval[1] = Math.max(newInterval[1], intervals[i][1]);
        i++;
    }

    ans.push(newInterval);

    while(i < n){
        ans.push(intervals[i++]);
    }

    return ans;
};

File: 0057-insert-interval/0057-insert-interval.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        int n = intervals.length;

        List<int[]> li = new ArrayList<>();

        int i = 0;

        while(i < n && intervals[i][1] < newInterval[0]){
            li.add(intervals[i]);
            i++;
        }


        while(i < n && intervals[i][0] <= newInterval[1]){
            newInterval[0] = Math.min(intervals[i][0], newInterval[0]);
            newInterval[1] = Math.max(intervals[i][1], newInterval[1]);
            i++;
        }

        li.add(newInterval);

        while(i < n){
            li.add(intervals[i++]);
        }

        int size = li.size();

        int[][] ans = new int[size][2];
        i = 0;

        for(int[] arr : li){
            ans[i][0] = arr[0];
            ans[i][1] = arr[1];
            i++;
        }

        return ans;
    }
}